lib/Nodes/TableNode.php

```diff
@@ -104,7 +108,14 @@ public function finalize(Parser $parser) : void
             }
 
             foreach ($row as &$col) {
-                $col = $parser->createSpan($col);
+                $lines       = explode("\n", (string) $col);
+                $lineChecker = new LineChecker(new LineDataParser());
```

Comment:
I would like to have this a dependency passed through instead of instantiating a new one here every time. But honestly, we sort of have the same issue already in DocumentParser where we create a new one there everytime. We can save this for another future round of refactoring.

Reply:
ok, you're right, i've passed it through the factory, then we're using the DocumentParser's instance